here-naksha-lib-core/src/test/java/com/here/naksha/lib/core/util/diff/PatcherTest.java

```diff
@@ -57,6 +59,88 @@ void basic() throws IOException {
+    assertTrue(nestedMapDiff34.get("isAdded") instanceof InsertOp);
+    assertTrue(nestedMapDiff34.get("willBeUpdated") instanceof UpdateOp);
+    assertTrue(nestedMapDiff34.get("willBeDeleted") instanceof RemoveOp);
+
+    // Assert nested array
+    final ListDiff nestedArrayDiff34 = (ListDiff) mapDiff34.get("array");
+    assertTrue(nestedArrayDiff34.get(1) instanceof UpdateOp);
+    assertTrue(nestedArrayDiff34.get(2) instanceof MapDiff);
+    // "retainedElement" is retained, does not appear in nestedMapDiff34
+    // InsertOp case for array (ListDiff) is addressed in the test testCompareSameArrayDifferentOrder()
+    assertTrue(nestedArrayDiff34.get(3) instanceof RemoveOp);
+
+    // Some extra nested JSON object in array assertions
+    assertTrue(((MapDiff) nestedArrayDiff34.get(2)).get("isAddedProperty") instanceof InsertOp);
+    assertTrue(((MapDiff) nestedArrayDiff34.get(2)).get("willBeDeletedProperty") instanceof RemoveOp);
+
+    // Modify the whole difference to get rid of all RemoveOp
+    mapDiff34.remove("firstToBeDeleted");
+    nestedMapDiff34.remove("willBeDeleted");
+    ((MapDiff) nestedArrayDiff34.get(2)).remove("willBeDeletedProperty");
+    nestedArrayDiff34.remove(3);
+
+    final JsonObject patchedf3 = Patcher.patch(f3,mapDiff34);
+    assertNotNull(patchedf3);
+    final JsonObject expectedPatchedf3 =
+            JsonSerializable.deserialize(IoHelp.readResource("patcher/feature_3_patched_to_4_no_remove.json"), JsonObject.class);
+    assertNotNull(expectedPatchedf3);
+
+    // Check that the patched feature 3 has the correct content
+    JSONAssert.assertEquals(patchedf3.serialize(),expectedPatchedf3.serialize(), JSONCompareMode.STRICT);
+    final Difference newDiff = Patcher.getDifference(patchedf3, expectedPatchedf3);
+    assertNull(newDiff);
+  }
+
+  @Test
+  void testCompareSameArrayDifferentOrder() {
+    final JsonObject f3 =
+            JsonSerializable.deserialize(IoHelp.readResource("patcher/feature_3.json"), JsonObject.class);
+    assertNotNull(f3);
+    final JsonObject f5 =
+            JsonSerializable.deserialize(IoHelp.readResource("patcher/feature_5.json"), JsonObject.class);
+    assertNotNull(f5);
+    final Difference diff35 = Patcher.getDifference(f3, f5);
+    assertNotNull(diff35);
+    assert (diff35 instanceof MapDiff);
+    final MapDiff mapDiff35 = (MapDiff) diff35;
+    assertEquals(1,mapDiff35.size());
+    assertTrue(mapDiff35.get("array") instanceof ListDiff);
+    final ListDiff nestedArrayDiff35 = (ListDiff) mapDiff35.get("array");
+    // The patcher compares array element by element in order,
+    // so the nested JSON in feature 3 is compared against the string in feature 5
+    // and the string in feature 3 is against the nested JSON in feature 5
+    assertTrue(nestedArrayDiff35.get(2) instanceof UpdateOp);
+    assertTrue(nestedArrayDiff35.get(3) instanceof UpdateOp);
+    assertTrue(nestedArrayDiff35.get(4) instanceof InsertOp);
```

Comment:
Good. This is important test case to validate array behaviour with mixed string and object comparison. Pls also apply patch and compare the result with expected outcome.

Reply:
roger, patching added :)